lib/recipe/bloc/recipe_bloc.dart

```diff
@@ -825,4 +853,163 @@ class RecipeBloc extends Bloc<RecipeEvent, RecipeState> {
+        }
+
+        if (recipeAttribute == 'Measurement') {
+          List<Measurement> measurements = List.from(recipe.measurements);
+
+          List<String> newMeasurements = recipeMeasurement.split('+');
+          for (String newMeasurement in newMeasurements) {
+            List<String> fields = newMeasurement.trim().split('_');
+            final measurement = Measurement(
+                measurementMethod: fields[0],
+                measurementUnit: fields[1],
+                measurementValue: fields[2]);
+            measurements.add(measurement);
+          }
+
+          recipe = recipe.copyWith(measurements: measurements);
+        } else if (recipeAttribute == 'Probe') {
+          List<Probe> probes = List.from(recipe.probes);
+
+          List<String> probeAnswers = recipeProbeAnswers.split('+');
+          List<Map<String, dynamic>> probeOptions = [];
+          for (String answer in probeAnswers) {
+            probeOptions
+                .add({'option': answer.trim(), 'id': const Uuid().v4()});
+          }
+          final probe =
+              Probe(probeName: recipeProbeName, probeOptions: probeOptions);
+          probes.add(probe);
+
+          recipe = recipe.copyWith(probes: probes);
+        } else if (recipeAttribute == 'Ingredient') {
+          List<Ingredient> ingredients = List.from(recipe.ingredients);
+
+          List<Measurement> measurements = [];
+          List<String> newMeasurements = ingredientMeasurement.split('+');
+          for (String newMeasurement in newMeasurements) {
+            List<String> fields = newMeasurement.trim().split('_');
+            final measurement = Measurement(
+                measurementMethod: fields[0],
+                measurementUnit: fields[1],
+                measurementValue: fields[2]);
+            measurements.add(measurement);
+          }
+          final ingredient = Ingredient(
+              name: ingredientName,
+              description: ingredientDescription,
+              cookingState: ingredientCookingState,
+              measurements: measurements,
+              saved: true); // TODO: Figure out food composition implementation
+          ingredients.add(ingredient);
+
+          recipe = recipe.copyWith(ingredients: ingredients);
+        } else {
+          continue; // Skip line with bad recipe attribute
+        }
+
+        newRecipes.add(recipe);
+      }
+
+      for (Recipe recipe in newRecipes) {
+        if (recipe.measurements.isEmpty) {
+          // Protect against recipes without Measurements
+          Recipe recipeWithMeasurement =
+              recipe.copyWith(measurements: [Measurement()]);
+          newRecipes.remove(recipe);
+          newRecipes.add(recipeWithMeasurement);
+        }
+      }
+
+      List<Recipe> recipes = existingRecipes + newRecipes;
+
+      emit(state.copyWith(
+          recipes: recipes,
+          recipeImportStatus:
+              'Imported ${newRecipes.length} recipe(s) successfully'));
```

Comment:
Maybe we can show how many were overwritten, and/or how many now imported successfully?

Reply:
This could be a todo for the future.